source_middleware.go

```diff
@@ -23,7 +37,13 @@ type SourceMiddleware interface {
 // all sources unless there's a good reason not to.
 func DefaultSourceMiddleware() []SourceMiddleware {
 	return []SourceMiddleware{
-		// TODO add default middleware (e.g. rate limit)
+		SourceWithSchema{
+			DefaultSchemaFormat:   schema.TypeAvro.String(),
```

Comment:
Unrelated to this PR (since I see the same pattern in other middleware): how about we name the fields without the prefix `Default`? When I'm reading a middleware's code, the prefix `Default` is a bit confusing. Here we see from the context that it's about default values. And also when we use it in the form `foo.defaults.SchemaFormat` it's also clear.

Reply:
I changed the field names, extracted the config into a separate type and added a way to configure the default middleware. This will make it easier to tweak the default middleware, previously it had to be done [awkwardly like this](https://github.com/conduitio-labs/conduit-connector-snowflake/blob/671448d3cb3718649b01a39447666cb6dac3c924/destination/destination.go#L50-L52). Let me know what you think, if we want to go with this I'll adjust the destination middleware to follow the same pattern.

Reply:
Ok seems like the pattern is set, in the next PR that introduces the destination middleware I'll apply it there as well.